src/shell/boards/board-container.tsx

```diff
@@ -73,9 +86,81 @@ const BoardDetailContainer = styled(Row)`
 const BackButton = styled(IconButton)``;
 const Actions = styled(Row)``;
 
+interface ListItemContentProps {
+	icon?: string;
+	label: string;
+	selected?: boolean;
+	app: string;
+	boardId: string;
+}
+
+function ListItemContent({
+	icon,
+	label,
+	selected,
+	app,
+	boardId
+}: ListItemContentProps): JSX.Element {
+	const t = getT();
+	const onClose = useCallback<IconButtonProps['onClick']>(
+		(ev) => {
+			ev.stopPropagation();
+			closeBoard(boardId);
+		},
+		[boardId]
+	);
+	return (
+		<Container orientation="horizontal" mainAlignment="flex-start">
+			{icon && (
+				<Padding right={'small'}>
+					<Icon icon={icon} size={'large'} color={'text'} style={{ pointerEvents: 'none' }} />
+				</Padding>
+			)}
+			<OverflowContainer crossAlignment={'flex-start'}>
+				<Text size={'medium'} weight={selected ? 'bold' : 'regular'} color={'gray0'}>
+					{label}
+				</Text>
+				<Text size={'small'} weight={selected ? 'bold' : 'regular'} color={'secondary'}>
+					{t('board.from_app', 'From {{app}}', {
+						app
+					})}
+				</Text>
+			</OverflowContainer>
+			<Padding left={'small'}>
+				<IconButton icon={'CloseOutline'} size={'large'} onClick={onClose} />
+			</Padding>
+		</Container>
+	);
+}
+
 export const BoardContainer: FC = () => {
 	const t = getT();
-	const { boards, minimized, expanded, current } = useBoardStore();
+	const { boards, minimized, expanded, current, orderedBoards } = useBoardStore();
+
+	const boardDropdownItems = useMemo(
+		(): DropdownItem[] =>
+			map(
+				orderedBoards,
+				(boardId): DropdownItem => ({
+					id: boardId,
+					label: boards[boardId].title,
+					icon: boards[boardId].icon,
+					onClick: () => setCurrentBoard(boardId),
+					selected: boardId === current,
+					customComponent: (
+						<ListItemContent
+							label={boards[boardId].title}
+							icon={boards[boardId].icon}
+							selected={boardId === current}
+							app={getApp(boards[boardId].app)().display}
```

Comment:
I'm afraid we need to use something else for the module name because this field is not translated :cry: